Tests/Blazorise.Tests/Components/DataGridComponentTest.cs

```diff
@@ -39,19 +45,330 @@ public void NoData_Should_Render_EmptyTemplate()
     public void SortByField_Should_CorrectlySortRows()
     {
         // setup
-        var expectedOrderedValues = new[] { "1/8", "1/4", "1/2", "3/4" };
+        var expectedOrderedValues = new[] { "1/8", "1/4", "1/3", "1/2", "3/4" };
 
         // test
         var comp = RenderComponent<DataGridComponent>();
-        var rows = comp.FindAll( "tbody tr td:nth-child(2)" );
 
         // validate
-        var count = 0;
-        foreach ( var item in rows )
+        comp.FindAll( "tbody tr td:nth-child(2)" )
+            .Select( x => x.TextContent )
+            .Should()
+            .BeEquivalentTo( expectedOrderedValues );
+    }
+
+    [Fact]
+    public void SortByField_Should_CorrectlyReorderRows()
+    {
+        // setup
+        var expectedOrderedValues = new[] { "3/4", "1/2", "1/3", "1/4", "1/8"  };
+        var comp = RenderComponent<DataGridComponent>();
+
+        // test
+        comp.Find( "thead tr th:nth-child(2)" )
+            .Click(); // change sort order to descending
+        
+        // validate
+        comp.FindAll( "tbody tr td:nth-child(2)" )
+            .Select( x => x.TextContent )
+            .Should()
+            .BeEquivalentTo( expectedOrderedValues );
+    }
+
+    [Fact]
+    public async Task SortByField_Should_RaiseSortChangedEvent()
+    {
+        // setup
+        var sortChanged = new List<DataGridSortChangedEventArgs>();
+        var comp = RenderComponent<DataGridComponent>( parameters =>
+        {
+            parameters.Add(
+                parameterSelector: x => x.SortChanged,
+                callback: e => sortChanged.Add( e ) );
+        } );
+
+        // test
+        await Task.Factory.StartNew(
```

Comment:
Can you explain briefly why we need all of these `Task.Factory.StartNew`

Reply:
During development, I often launch the existing unit tests. It seems that there are currently some "unstable" tests. My colleagues call these tests "blink tests" because they constantly alternate between green (ok) and red (fail).

The experience we've had with our own bUnit tests is that these tests are often timing issues of async/parallel code.

In *DataGrid.razor.cs* there are a few methods that call `InvokeAsync(..)` without awaiting the method call (e.g. the events that are raised inside `HandleSortColumn` -I haven't investigated why this method returns `void` instead of `Task`). 

Not awaiting the `InvokeAsync` method results in parallel processing or out-of-order processing. You cannot write a stable unit test without having control over when **a method actually finishes executing**.

I thought about how to get synchronous method execution. The only idea I had was to write my own TaskScheduler. It's called `CurrentThreadTaskScheduler` and has a maximum concurrency level of 1. It doesn't allow scheduling of tasks and (hopefully) enforces synchronous execution.

Some examples of currently unstable tests:
![image](https://user-images.githubusercontent.com/3825484/224082084-d05315f3-3312-4326-85ca-f6b8d8f82165.png)
![image](https://user-images.githubusercontent.com/3825484/224082123-10dc42d3-322a-4112-abf2-212c95e9d072.png)
![image](https://user-images.githubusercontent.com/3825484/224082160-6c592f86-d8f6-40c3-9363-2fa54b53e249.png)

Reply:
What do you think about this "hack"?

Reply:
Fair enough, you're right we've been having our issues with bunit also due to async/parallel code and therefore unstable/unreliable tests.
I think in some places we just opted for retry for expected state, or wait longer, but this seems like at least an alternative approach. Thank you.